plugins/source/gcp/codegen/recipes/compute.go

```diff
@@ -77,135 +65,109 @@ var computeResourcesAggList = []*Resource{
-		RequestStruct:  &pb.AggregatedListVpnGatewaysRequest{},
-		ResponseStruct: &pb.VpnGatewayAggregatedList{},
-		ListFunction:   (&compute.VpnGatewaysClient{}).AggregatedList,
-		OutputField:    "Value.VpnGateways",
-		Description:    "https://cloud.google.com/compute/docs/reference/rest/v1/vpnGateways#VpnGateway",
+		SubService:   "vpn_gateways",
+		Struct:       &pb.VpnGateway{},
+		NewFunction:  compute.NewVpnGatewaysRESTClient,
+		ListFunction: (&compute.VpnGatewaysClient{}).AggregatedList,
+		OutputField:  "Value.VpnGateways",
+		Description:  "https://cloud.google.com/compute/docs/reference/rest/v1/vpnGateways#VpnGateway",
 	},
 	{
-		SubService:     "instance_groups",
-		Struct:         &pb.InstanceGroup{},
-		NewFunction:    compute.NewInstanceGroupsRESTClient,
-		RequestStruct:  &pb.AggregatedListInstanceGroupsRequest{},
-		ResponseStruct: &pb.InstanceGroupAggregatedList{},
-		ListFunction:   (&compute.InstanceGroupsClient{}).AggregatedList,
-		OutputField:    "Value.InstanceGroups",
-		Description:    "https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups#InstanceGroup",
+		SubService:   "instance_groups",
+		Struct:       &pb.InstanceGroup{},
+		NewFunction:  compute.NewInstanceGroupsRESTClient,
+		ListFunction: (&compute.InstanceGroupsClient{}).AggregatedList,
+		OutputField:  "Value.InstanceGroups",
+		Description:  "https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups#InstanceGroup",
 	},
 }
 
 var computeResourcesList = []*Resource{
 	{
-		SubService:     "images",
-		Struct:         &pb.Image{},
-		NewFunction:    compute.NewImagesRESTClient,
-		RequestStruct:  &pb.ListImagesRequest{},
-		ResponseStruct: &pb.ImageList{},
-		ListFunction:   (&compute.ImagesClient{}).List,
+		SubService:   "images",
+		Struct:       &pb.Image{},
+		NewFunction:  compute.NewImagesRESTClient,
+		ListFunction: (&compute.ImagesClient{}).List,
 	},
 	{
-		SubService:     "firewalls",
-		Struct:         &pb.Firewall{},
-		NewFunction:    compute.NewFirewallsRESTClient,
-		RequestStruct:  &pb.ListFirewallsRequest{},
-		ResponseStruct: &pb.FirewallList{},
-		ListFunction:   (&compute.FirewallsClient{}).List,
+		SubService:   "firewalls",
+		Struct:       &pb.Firewall{},
+		NewFunction:  compute.NewFirewallsRESTClient,
+		ListFunction: (&compute.FirewallsClient{}).List,
 	},
 	{
 		SubService:      "networks",
 		Struct:          &pb.Network{},
 		NewFunction:     compute.NewNetworksRESTClient,
-		RequestStruct:   &pb.ListNetworksRequest{},
-		ResponseStruct:  &pb.NetworkList{},
 		ListFunction:    (&compute.NetworksClient{}).List,
 		NameTransformer: CreateReplaceTransformer(map[string]string{"ipv_6": "ipv6", "i_pv4": "ipv4"}),
 	},
 	{
 		SubService:     "ssl_policies",
 		Struct:         &pb.SslPolicy{},
 		NewFunction:    compute.NewSslPoliciesRESTClient,
-		RequestStruct:  &pb.ListSslPoliciesRequest{},
 		ResponseStruct: &pb.SslPoliciesList{},
-		ListFunction:   (&compute.InterconnectsClient{}).List,
```

Comment:
This is wrong, we should be using `SslPoliciesClient`. It doesn't impact the current codegen as it only uses the function name `List`

Reply:
But why is it wrong...